ndt7/server/results/file.go

```diff
@@ -16,15 +18,19 @@ type File struct {
 	// Writer is the gzip writer instance
 	Writer *gzip.Writer
 	// Fp is the underlying file
-	Fp     *os.File
+	Fp *os.File
 }
 
 // newFile opens a measurements file in the current working
 // directory on success and returns an error on failure.
-func newFile() (*File, error) {
-	format := "2006-01-02T15:04:05.000000000Z"
-	timestamp := time.Now().UTC().Format(format)
-	name := "ndt7-" + timestamp + ".jsonl.gz"
+func newFile(datadir, what, uuid string) (*File, error) {
+	timestamp := time.Now().UTC()
+	dir := path.Join(datadir, what, timestamp.Format("2006/01/02"))
```

Comment:
Why do you separate download and upload at the dir level? (I'd have put that in the filename, perhaps)